modules/vaos/spec/controllers/v2/appointments_controller_spec.rb

```diff
@@ -163,4 +163,188 @@
+        expect do
+          controller.send(:get_provider_appointment_type_id, provider)
+        end.to raise_error(Common::Exceptions::BackendServiceException) { |e|
+          expect(e).to have_attributes(
+            key: 'PROVIDER_SELF_SCHEDULABLE_TYPES_MISSING',
+            original_status: 502,
+            original_body: 'No self-schedulable appointment types available for this provider'
+          )
+        }
+      end
+    end
+
+    context 'when provider has self-schedulable appointment types' do
+      let(:provider) do
+        OpenStruct.new(
+          appointment_types: [
+            { id: '1', is_self_schedulable: false },
+            { id: '2', is_self_schedulable: true },
+            { id: '3', is_self_schedulable: true }
+          ]
+        )
+      end
+
+      it 'returns the ID of the first self-schedulable appointment type' do
+        result = controller.send(:get_provider_appointment_type_id, provider)
+        expect(result).to eq('2')
+      end
+    end
+
+    context 'when provider has mixed appointment types with different is_self_schedulable values' do
+      let(:provider) do
+        OpenStruct.new(
+          appointment_types: [
+            { id: '1' }, # missing property
+            { id: '2', is_self_schedulable: nil },
+            { id: '3', is_self_schedulable: false },
+            { id: '4', is_self_schedulable: true },
+            { id: '5', is_self_schedulable: true }
+          ]
+        )
+      end
+
+      it 'returns the ID of the first appointment type where is_self_schedulable is explicitly true' do
+        result = controller.send(:get_provider_appointment_type_id, provider)
+        expect(result).to eq('4')
+      end
+    end
+  end
+
+  describe '#fetch_provider_slots' do
+    let(:controller) { described_class.new }
+    let(:eps_provider_service) { instance_double(Eps::ProviderService) }
+    let(:referral) do
+      OpenStruct.new(
+        referral_date: '2024-01-01',
+        expiration_date: '2024-12-31'
+      )
+    end
+    let(:provider) do
+      OpenStruct.new(
+        id: 'provider123',
+        appointment_types: [{ id: 'type123', is_self_schedulable: true }]
+      )
+    end
+
+    before do
+      allow(controller).to receive(:eps_provider_service).and_return(eps_provider_service)
+    end
+
+    context 'when provider has valid self-schedulable appointment types' do
+      let(:expected_slots) { [{ id: 'slot1' }, { id: 'slot2' }] }
+
+      before do
+        allow(eps_provider_service).to receive(:get_provider_slots).and_return(expected_slots)
+      end
+
+      it 'calls get_provider_slots with correct parameters' do
+        result = controller.send(:fetch_provider_slots, referral, provider)
+
+        expect(eps_provider_service).to have_received(:get_provider_slots).with(
+          'provider123',
+          hash_including(
+            appointmentTypeId: 'type123',
+            startOnOrAfter: a_string_starting_with('2024-01-01T'),
```

Comment:
Note: This expectation checks only the date as the date/time conversion adds time based on the system's time zone. Locally that would be 05:00 for my ET laptop, but the CI system will be 00:00Z.